src/arithmetic/arithmetic_expression.c

```diff
@@ -315,23 +315,25 @@ void AR_EXP_Free(AR_ExpNode *root) {
 /* Mathematical functions - numeric */
 
 SIValue AR_ADD(SIValue *argv, int argc) {
-    SIValue result;
-    result = argv[0];
+    SIValue result = argv[0];
+    /* If the result is numeric, ensure that it is represented as a double. */
+    bool numeric_result = SIValue_ConvertToDouble(&result);
     char buffer[512];
     char *string_arg = NULL;
+    double numeric_arg;
 
     for(int i = 1; i < argc; i++) {
         if(SIValue_IsNull(argv[i])) return SI_NullVal();
 
         /* Perform numeric addition only if both result and current argument
          * are numeric. */
-        if(SI_TYPE(result) & SI_NUMERIC && SI_TYPE(argv[i]) & SI_NUMERIC) {
+        if(numeric_result && SIValue_ToDouble(&argv[i], &numeric_arg)) {
             /* Numeric addition. */
-            result.doubleval += argv[i].doubleval;
+            result.doubleval += numeric_arg;
```

Comment:
I think we can simplify this:
Leave the original if condition, and update this assignment

```
SIValue result = argv[0];
if(SI_TYPE(result) & SI_NUMERIC)
SIValue_ConvertToDouble(&result);
```
...
```
if(SI_TYPE(result) & SI_NUMERIC && SI_TYPE(argv[i]) & SI_NUMERIC) {	
SIValue_ToDouble(&argv[i], &numeric_arg);
result.doubleval += numeric_arg;
}
```

Reply:
:+1: